ale/base/data_isis.py

```diff
@@ -36,187 +37,144 @@ def read_table_data(table_label, cube):
-
-    Returns
-    -------
-    str :
-        The format string for the entry binary
-    """
-    data_formats = {
-        'Integer' : 'i',
-        'Double'  : 'd',
-        'Real'    : 'f'
-    }
-    return data_formats[field_label['Type']] * field_label['Size']
-
-def parse_field(field_label, data, encoding='latin_1'):
-    """
-    Parses a binary table field entry and converts it into
-    an in memory data type
-
-    Parameters
-    ----------
-    field_label : PVLModule
-                  The field label
-
-    data : bytes
-           The binary data for the field entry
-
-    Returns
-    -------
-    Union[int, float, str, list] :
-        The table field entry converted to a native python
-        type
-    """
-    if field_label['Type'] == 'Text':
-        field_data = data[:field_label['Size']].decode(encoding=encoding)
-    else:
-        data_format = field_format(field_label)
-        field_data = struct.unpack_from(data_format, data)
-        if len(field_data) == 1:
-            field_data = field_data[0]
-    return field_data
-
-def parse_table_data(table_label, data):
-    """
-    Parses an ISIS table into a dict where the keys are the
-    field names and the values are lists of entries.
+    Parse an ISIS table into a dictionary.
 
     Parameters
     ----------
     table_label : PVLModule
-                  The table label
-
+                  The ISIS table label
     data : bytes
-           The binary data for the entire table
+           The binary component of the ISIS table
 
     Returns
     -------
     dict :
-        The table as a dict
+           The table as a dictionary with the keywords from the label and the
+           binary data
     """
+    data_sizes = {'Integer' : 4,
+                  'Double'  : 8,
+                  'Real'    : 4,
+                  'Text'    : 1}
+    data_formats = {'Integer' : 'i',
+                    'Double'  : 'd',
+                    'Real'    : 'f'}
+
+    # Parse the binary data
     fields = table_label.getlist('Field')
     results = {field['Name']:[] for field in fields}
     offset = 0
     for record in range(table_label['Records']):
         for field in fields:
-            field_data = parse_field(field, data[offset:])
+            if field['Type'] == 'Text':
+                field_data = data[offset:offset+field['Size']].decode(encoding='latin_1')
+            else:
+                data_format = data_formats[field['Type']] * field['Size']
+                field_data = struct.unpack_from(data_format, data[offset:])
+                if len(field_data) == 1:
+                    field_data = field_data[0]
+
             results[field['Name']].append(field_data)
-            offset += field_size(field)
-    return results
+            offset += data_sizes[field['Type']] * field['Size']
+
+    # Parse the keywords from the label
+    results.update({key : value for key, value in table_label.items() if not isinstance(value, pvl._collections.PVLGroup)})
```

Comment:
Seems like this function never returns. Not sure if this is the intended behavior but it seems incorrect